core/src/main/java/org/opensearch/sql/executor/pagination/PaginatedPlanCache.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.executor.pagination;
+
+import com.google.common.hash.HashCode;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+import lombok.RequiredArgsConstructor;
+import org.opensearch.sql.ast.tree.UnresolvedPlan;
+import org.opensearch.sql.expression.NamedExpression;
+import org.opensearch.sql.expression.serialization.DefaultExpressionSerializer;
+import org.opensearch.sql.planner.physical.PaginateOperator;
+import org.opensearch.sql.planner.physical.PhysicalPlan;
+import org.opensearch.sql.planner.physical.ProjectOperator;
+import org.opensearch.sql.storage.StorageEngine;
+import org.opensearch.sql.storage.TableScanOperator;
+
+/**
+ * This class is entry point to paged requests. It is responsible to cursor serialization
+ * and deserialization.
+ */
+@RequiredArgsConstructor
+public class PaginatedPlanCache {
```

Comment:
Why we call it Cache?

Reply:
What do you think of `PlanSerializer`?

Originally we considered caching plans with open cursors in a system index, hence the name -- paginated plan cache.

Reply:
Fixed in 64386f30d.